### PR TITLE
8208169: can not print selected pages of web page

### PR DESCRIPTION
**Pull request by bhaweshkc**
Print function of WebEngine.java ignores page range setting and prints given number of pages starting from first page, which is the root cause of this issue. To fix it, put check for page ranges and if it available, use it for printing pages otherwise print all pages as usual.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8208169](https://bugs.openjdk.java.net/browse/JDK-8208169): can not print selected pages of web page

### Reviewers
 * Phil Race ([prr](@prrace) - **Reviewer**)
 * Kevin Rushforth ([kcr](@kevinrushforth) - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jfx pull/222/head:pull/222`
`$ git checkout pull/222`

### CODE REVIEW

**Comment by kevinrushforth** [comment]
@prrace can you also review this?

**Comment by kevinrushforth** [comment]
Can you add a test for this? It may need to be a manual test, in which case it can go into `tests/manual/printing/`.

File: tests/manual/printing/PrintPageRangeTest.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    static final String instructions =
+                    " 1. Press print button to open Print Dialog.\n" +
+                    " 2. Select a Printer and PageRange 3 to 4 in page range selection.\n" +
+                    " 3. Click Ok.\n" +
+                    " 4. After this 2 pages must be printed.\n" +
+                    " 5. When all pages are printed you will see <END OF PRINT JOB> on bottom.\n" +
+                    " 6. Check whether printed pages are page no 3 and page no 4.\n" +
+                    " 7. if Yes then Test Passed else Test Failed";
+
+    static final String initialURL = "https://en.wikipedia.org/wiki/Java_version_history";
```
Comment (prrace) [inline tests/manual/printing/PrintPageRangeTest.java:78]: Have you tested this when using a system proxy ? And do the webview tests in general use random webpages from the internet ? Why can't we generate some simple local content that we KNOW is 4 pages long. You don't even know that for sure about this page.

**Comment by bhaweshkc** [comment]
@prrace 
updated test which uses locally generated html content which should be large enough to cover 4 pages.

File: tests/manual/printing/PrintPageRangeTest.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                    " 1. Press print button to open Print Dialog.\n" +
+                    " 2. Select a Printer, Page Size A4 and PageRange 3 to 4 in page range selection.\n" +
+                    " 3. Click Ok.\n" +
+                    " 4. After this 2 pages must be printed.\n" +
+                    " 5. When all pages are printed you will see <END OF PRINT JOB> on bottom.\n" +
+                    " 6. Check whether printed pages are page no 3 and page no 4.\n" +
+                    "     Note: In printed pages, first page should start with approx <HTML Line No. 50>.\n" +
+                    " 7. if Yes then Test Passed else Test Failed";
+
+    static final String initialURL = "https://en.wikipedia.org/wiki/Java_version_history";
```
Comment (arun-Joseph) [inline tests/manual/printing/PrintPageRangeTest.java:80]: initialURL is no longer required

File: modules/javafx.web/src/main/java/javafx/scene/web/WebEngine.java [hunk 3]
```diff
@@ -1609,10 +1611,23 @@ public void print(PrinterJob job) {
         float height = (float) pl.getPrintableHeight();
         int pageCount = page.beginPrinting(width, height);
 
-        for (int i = 0; i < pageCount; i++) {
-            if (printStatusOK(job)) {
-                Node printable = new Printable(page, i, width);
-                job.printPage(printable);
+        JobSettings jobSettings = job.getJobSettings();
+        if(jobSettings.getPageRanges() != null) {
```
Comment (kevinrushforth) [inline modules/javafx.web/src/main/java/javafx/scene/web/WebEngine.java:1615]: Minor: space after `if`.

File: tests/manual/printing/PrintPageRangeTest.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                    " 4. After this 2 pages must be printed.\n" +
+                    " 5. When all pages are printed you will see <END OF PRINT JOB> on bottom.\n" +
+                    " 6. Check whether printed pages are page no 3 and page no 4.\n" +
+                    "     Note: In printed pages, first page should start with approx <HTML Line No. 50>.\n" +
+                    " 7. if Yes then Test Passed else Test Failed";
+
+    private String createHtmlPage() {
+        StringBuilder htmlStringBuilder = new StringBuilder();
+        htmlStringBuilder.append("<html><head></head><body>");
+        for(int i = 0; i < 500; ++i) {
```
Comment (kevinrushforth) [inline tests/manual/printing/PrintPageRangeTest.java:83]: Minor: space after `for`

File: tests/manual/printing/PrintPageRangeTest.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        final VBox root = new VBox();
+        root.getChildren().addAll(instructionsText, naviBar, webView, bottomMessageLabel);
+        VBox.setVgrow(webView, Priority.ALWAYS);
+        String htmlContent = createHtmlPage();
+        webEngine.loadContent(htmlContent);
+        scene = new Scene(root);
+        return scene;
+    }
+
+    private void SetMessage(String msg) {
```
Comment (kevinrushforth) [inline tests/manual/printing/PrintPageRangeTest.java:122]: Minor: usual naming convention is for an initial lower-case letter.

**Pull request closed without merging**